ui/src/message_popup/renderers/minimal_timed_view.jsx

```diff
@@ -0,0 +1,88 @@
+/* @flow weak */
+import React from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+
+var sceneTimer;
+
+// This response supports responding with text.
+export default React.createClass({
+  displayName: 'MinimalTimedView',
+
+  propTypes: {
+    onLogMessage: React.PropTypes.func.isRequired,
+    onResponseSubmitted: React.PropTypes.func.isRequired,
+    recordText: React.PropTypes.string,
+  },
+
+  getDefaultProps() {
+    return {
+      recordText: 'Respond',
+      autoFocus: true,
+      textHeight: 48
+    };
+  },
+
+
+  getInitialState() {
+    sceneTimer = setTimeout(this.onTimer, 25000);
```

Comment:
Nice!  This is exactly the essence of how to do this. 👍 

There's three small details here about working with timers, which involve a bit more about how React component lifecycles work.

First, instead of using a variable at the top of the file (`var sceneTimer`), let's define this as an instance variable that's part of the React component.  The variable at the top is "closed over" any time this code is run, and so there isn't a unique `sceneTimer` for each `MinimalTimedView` that is created if the page happened to have multiple instances of that component.  That doesn't really matter for our use case, but in general if you have a component or a class, you want to encapsulate its behavior as an instance variable (or in React as state) so that each instance has its own copy.  Concretely, remove the `var` declaration up top, and use `this.sceneTimer = ...` to attach that variable value to the particular React component instance that's created.  You can then read that value back with `this.sceneTimer`.

Second, calling `setTimeout` is a "side effect," meaning that calling it doesn't return a number or a string or a value but it causes something else to happen.  In React, the place for doing that is `componentDidMount` and not `getInitialState`, which should just return a value and not causing anything to happen (like making a server request or printing anything to the console or scheduling any timers).  So just add `componentDidMount` and move this there.

Third, when this component is mounted on the page, the timer starts.  What happens if something else gets rendered instead?  Like, say the user clicks on the reset button in the corner and restarts the scenario altogether?  Right now, nothing would cancel that timer, and it would still run 25 seconds later.  It would then either try to call log message and probably lead to some errors.  You've already taken care of this for the particular case when the user clicks the skip button, but whenever there are "side effects" or timers or other things that we set up in `componentDidMount`, the place to tear those down or clean them up is in `componentWillUnmount`.  That way it'll get cleaned up no matter what happens when this component is unmounted and removed from the screen, however that happens.  So concretely, add `componentWillUnmount` and move the `clearTimeout` into there.  If skip is clicked, it will lead to rendering something different on the page, and then `componentWillUnmount` will be called and the timer will be cleared then.

More on React lifecycle methods here if you're curious: https://facebook.github.io/react/docs/react-component.html.